File: packages/skill-eval-python/src/skill_eval/schemas/grading.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ExpectationResult(BaseModel):
    text: str
    passed: bool
    evidence: str


class GradingResult(BaseModel):
    pass_rate: float = Field(ge=0, le=1)
    passed: int = Field(ge=0)
    failed: int = Field(ge=0)
    total: int = Field(ge=1)
    expectations: list[ExpectationResult]
    claims: list[str] = Field(default_factory=list)
    eval_feedback: str | None = None

File: packages/skillet/src/eval/schemas/grading.ts
import { z } from "zod";

export const ExpectationResultSchema = z.object({
	text: z.string(),
	passed: z.boolean(),
	evidence: z.string(),
});

export const GradingResultSchema = z.object({
	pass_rate: z.number().min(0).max(1),
	passed: z.number().int().min(0),
	failed: z.number().int().min(0),
	total: z.number().int().min(1),
	expectations: z.array(ExpectationResultSchema),
	claims: z.array(z.string()).default([]),
	eval_feedback: z.string().nullable().default(null),
});

export type ExpectationResult = z.infer<typeof ExpectationResultSchema>;
export type GradingResult = z.infer<typeof GradingResultSchema>;
